question.java

```diff
@@ -37,9 +37,28 @@
     //     Merge two sorted arrays.
     //     arr1 and arr2 are two sorted arrays and are passed in as parameters.
     //     Merge arr1[0..n1-1] and arr2[0..n2-1] and return a sorted array [0..n1+n2-1] 
-    public static void mergeArrays(int[] arr1, int[] arr2) 
+    public static int[] mergeArrays(int[] arr1, int[] arr2) 
     { 
-      return;
+        int len = arr1.length + arr2.length;
+        int[] arr3 = new int[len];
+
+        i1 = 0;
+        i2 = 0;
+        while (i1 + i2 < len) {
+            if (i1 == arr1.length) {
+                arr3[i1 + i2] = arr2[i2];
```

Comment:
What if arr2 is longer than arr1? Since you don't change the value of i1 and i2 here, you might get into infinite loop!

Reply:
This branch will be hit in the case that arr2 is greater than arr1 since i1 will have gone through all values of arr1 and only i2 will then be incremented.

Reply:
Yeah but i2 isn't actually incremented here.
Consider this:
arr1[1]
arr2[3, 4, 5, 6, 7, 8, 9, 10]
After first iteration, i1 = 1; i2 = 0.
In second iteration, since i1 = 1 = arr.length, arr[i1 + i2] (arr[1]) = arr2[0].
So is the third, the forth, etc.
Hence, infinite loop

Reply:
Be sure to be looking at the latest commit

Reply:
Oh I see.
In that case, I think your solution looks good!